galata/test/jupyterlab/sidebars.test.ts

```diff
@@ -194,3 +201,157 @@ test.describe('Sidebars', () => {
     expect(tableOfContentsElementRole).toEqual('region');
   });
 });
+
+const runningSessionsElementAriaLabels: string[] = [
+  'Open Tabs Section',
+  'Kernels Section',
+  'Language servers Section',
+  'Terminals Section'
+];
+const debuggerElementAriaLabels: string[] = [
+  'Variables Section',
+  'Callstack Section',
+  'Breakpoints Section',
+  'Source Section',
+  'Kernel Sources Section'
+];
+const propertyElementAriaLabels: string[] = [
+  'Warning Section',
+  'Installed Section',
+  'Discover Section'
+];
+
+const sessionsAndDebuggerAriaLabels: string[] =
+  runningSessionsElementAriaLabels.concat(debuggerElementAriaLabels);
+
+test.describe('Sidebar keyboard navigation @a11y', () => {
+  leftSidebarIds.forEach(leftSidebarId => {
+    test(`Open Sidebar tab ${leftSidebarId} via keyboard navigation`, async ({
+      page
+    }) => {
+      await page.goto();
+
+      await page.sidebar.close('left');
+
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
+        await page.keyboard.press('Tab');
+        let IsFocused = await page.evaluate(
+          () => document.activeElement?.getAttribute('data-id')
+        );
+        if (IsFocused === leftSidebarIds[0]) {
+          break;
+        }
+      }
```

Comment:
Since this pattern can be found several time in the tests, it would be nice to have a generic function to navigate using keyboard as helper, maybe in [activity helper](https://github.com/jupyterlab/jupyterlab/blob/main/galata/src/helpers/activity.ts).
That way it will be available for futur navigation tests.

Reply:
Hi @brichet, thanks for your comments. I've implemented most however, I'm struggling to get the the left/right side bar test passing with use of a helper function. Would you be able to advise me on where I'm going wrong in implementing this?

Thanks

Reply:
@e218736 you should try to run the tests locally, it would be easier to debug.
You need 2 terminals in your develop environment, and in the directory `galata`.
In the first one, you run `jlpm start` to start jupyterlab for the tests.
In the second one you run 
- `jlpm test test/jupyterlab/sidebars.test.ts` to test the whole file
- or `jlpm test test/jupyterlab/sidebars.test.ts -g 'via keyboard navigation'` (for example) to test only some specific test (using regex).

In this specific case, you are not sending the correct `dataId`.
`sideBarTabName` is the `data-id` (a string). So `sideBarTabName[0]` is only the first letter of the string.

To make it more generic, you can also add a parameter `attributeName` to the function, to be able to use it also with the "*open accordion panel*" tests.

Reply:
Hi @brichet,

Apologies but I'm unable to run tests locally. 

Thank you for the comments, I've amended if you are able to review again please.